File: files/versions/С/scr0020.c
#include <windows.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <time.h>
#include <psapi.h>

#pragma comment(lib, "psapi.lib")
#pragma comment(lib, "user32.lib")

#define VERSION "0.020"
#define FILENAME "u.txt"

HANDLE hConsole;

void cls() {
    system("cls");
}

void setColor(int color) {
    SetConsoleTextAttribute(hConsole, color);
}

void writeUsername(const char* name) {
    FILE* f = fopen(FILENAME, "w");
    if (f) {
        fprintf(f, "%s", name);
        fclose(f);
    }
}

int readUsername(char* buffer, int size) {
    FILE* f = fopen(FILENAME, "r");
    if (f) {
        if (fgets(buffer, size, f)) {
            size_t len = strlen(buffer);
            if (len > 0 && buffer[len-1] == '\n') buffer[len-1] = '\0';
            fclose(f);
            return 1;
        }
        fclose(f);
    }
    return 0;
}

void beep() {
    Beep(700, 500);
}

void beeep() {
    Beep(900, 300);
    Beep(1300, 300);
}

int main() {
    char name[256];
    char cmd[256];
    char newname[256];
    char text[256];
    char aquest[10];
    
    hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
    
    if (!readUsername(name, sizeof(name))) {
        setColor(10);
        printf("get.username:;\n");
        Sleep(500);
        printf("get.username = err:\n");
        Sleep(100);
        printf("start(), get:\n");
        Sleep(50);
        printf("start - True:\n");
        Sleep(100);
        printf("check.startlist:\n");
        Sleep(500);
        printf("get.all.startlist:\n");
        Sleep(100);
        printf("startlist:\ninput('username')\nafter.input('username')\nlambda.startlist(get(True)))\n");
        Sleep(100);
        printf("lambda: run.startlist.directly = True;\n");
        Sleep(100);
        cls();
        
        while (1) {
            setColor(10);
            printf("l: ");
            setColor(7);
            printf("enter your username\n");
            Beep(1000, 300);
            Beep(1300, 400);
            setColor(14);
            printf("lingua: ");
            setColor(7);
            fgets(name, sizeof(name), stdin);
            name[strcspn(name, "\n")] = 0;
            
            if (strlen(name) > 0) {
                writeUsername(name);
                break;
            } else {
                setColor(12);
                printf("err: ");
                setColor(7);
                printf("username not defined\n");
                Beep(500, 500);
            }
        }
    }
    
    cls();
    setColor(7);
    printf("hi, %s!\n", name);
    Beep(1100, 450);
    Beep(1300, 700);
    
    while (1) {
        setColor(14);
        printf("jungua");
        setColor(10);
        printf("@");
        setColor(7);
        printf("%s: ", name);
        fgets(cmd, sizeof(cmd), stdin);
        cmd[strcspn(cmd, "\n")] = 0;
        Beep(1500, 200);
        
        if (strcmp(cmd, "whoami") == 0) {
            setColor(14);
            printf("your username: ");
            setColor(7);
            printf("%s\n", name);
        }
        else if (strcmp(cmd, "exit") == 0) {
            break;
        }
        else if (strcmp(cmd, "fetch") == 0) {
            MEMORYSTATUSEX memStatus;
            memStatus.dwLength = sizeof(memStatus);
            GlobalMemoryStatusEx(&memStatus);
            
            DWORD drives = GetLogicalDrives();
            char drivePath[] = "C:\\";
            ULARGE_INTEGER freeBytes, totalBytes;
            GetDiskFreeSpaceExA(drivePath, &freeBytes, &totalBytes, NULL);
            
            double diskgb = (double)totalBytes.QuadPart / (1024.0 * 1024.0 * 1024.0);
            double ramgb = (double)memStatus.ullTotalPhys / (1024.0 * 1024.0 * 1024.0);
            
            time_t now;
            time(&now);
            struct tm *local = localtime(&now);
            
            setColor(14);
            printf("\n             ###########\n");
            printf("              #########\n");
            printf("              #########\n");
            printf("             ###########\n");
            printf("             \n");
            printf("             ###########\n");
            printf("              #########\n");
            printf("              #########\n");
            printf("              #########\n");
            printf("              #########\n");
            printf("    ######    #########\n");
            printf("   ######## ##########\n");
            printf("     ################\n");
            printf("        #########\n\n");
            
            setColor(7);
            setColor(14);
            printf("version: ");
            setColor(7);
            printf("%s\n", VERSION);
            setColor(14);
            printf("disk storage: ");
            setColor(7);
            printf("%.2fGB\n", diskgb);
            setColor(14);
            printf("ram storage: ");
            setColor(7);
            printf("%.2fGB\n", ramgb);
            setColor(14);
            printf("time now: ");
            setColor(7);
            printf("%04d-%02d-%02d %02d:%02d:%02d\n", 
                   local->tm_year + 1900, local->tm_mon + 1, local->tm_mday,
                   local->tm_hour, local->tm_min, local->tm_sec);
        }
        else if (strcmp(cmd, "clear") == 0 || strcmp(cmd, "cls") == 0) {
            cls();
        }
        else if (strncmp(cmd, "say(", 4) == 0 && cmd[strlen(cmd)-1] == ')') {
            strcpy(text, cmd + 4);
            text[strlen(text)-1] = '\0';
            setColor(10);
            printf("%s\n", text);
            setColor(7);
        }
        else if (strcmp(cmd, "rename") == 0) {
            beeep();
            setColor(10);
            printf("rename: ");
            setColor(7);
            fgets(newname, sizeof(newname), stdin);
            newname[strcspn(newname, "\n")] = 0;
            
            if (strlen(newname) == 0) {
                setColor(12);
                printf("err: ");
                setColor(7);
                printf("username not defined\n");
                Beep(500, 500);
                continue;
            }
            
            beep();
            setColor(12);
            printf("do you really want to rename?\n");
            setColor(14);
            printf("before rename: ");
            setColor(7);
            printf("%s\n", name);
            setColor(14);
            printf("after rename: ");
            setColor(7);
            printf("%s\n", newname);
            setColor(8);
            printf("(y/n) ");
            setColor(7);
            fgets(aquest, sizeof(aquest), stdin);
            
            if (aquest[0] == 'y' || aquest[0] == 'Y') {
                if (strcmp(newname, name) == 0) {
                    setColor(14);
                    printf("info: ");
                    setColor(7);
                    printf("this is already your name\n");
                    Beep(800, 300);
                    continue;
                } else {
                    writeUsername(newname);
                    strcpy(name, newname);
                    Beep(1000, 300);
                    Beep(1200, 300);
                }
            } else {
                setColor(14);
                printf("rename cancelled\n");
                setColor(7);
                Beep(600, 300);
            }
        }
        else if (strcmp(cmd, "help") == 0) {
            beep();
            setColor(12);
            printf("check ");
            setColor(14);
            printf("'README.md' ");
            setColor(12);
            printf("on official repository: ");
            setColor(8);
            printf("https://github.com/arxxdd-scheg/jungua\n");
            setColor(7);
        }
    }
    
    return 0;
}

File: files/versions/scripts/scr0020.py
import winsound
import time
import threading
import psutil
import os
import socket
import GPUtil
from colorama import Fore
def cls():
    os.system('cls')
def write(name):
    with open("u.txt", "w", encoding="utf-8") as f:
        f.write(name)
def beep():
    winsound.Beep(700, 500)
def beeep():
    winsound.Beep(900, 300)
    winsound.Beep(1300, 300)

if os.path.exists("u.txt") and os.path.getsize("u.txt") > 0:
    with open("u.txt", "r", encoding="utf-8") as file:
        name = file.read().strip()
else:
    print("get.username:;")
    time.sleep(0.5)
    print("get.username = err:")
    time.sleep(0.1)
    print("start(), get:")
    time.sleep(0.05)
    print("start - True:")
    time.sleep(0.1)
    print("check.startlist:")
    time.sleep(0.5)
    print("get.all.startlist:")
    time.sleep(0.1)
    print("startlist:\ninput('username')\nafter.input('username')\nlambda.startlist(get(True)))")
    time.sleep(0.1)
    print("lambda: run.startlist.directly = True;")
    time.sleep(0.1)
    cls()
    while True:
        print(Fore.GREEN + "l: " + Fore.WHITE + "enter your username")
        winsound.Beep(1000, 300)
        winsound.Beep(1300, 400)
        name = input(Fore.YELLOW + "lingua: " + Fore.WHITE)
        if name:
            write(name)
            break
        else:
            print(Fore.RED + "err: " + Fore.WHITE + "username not defined")
            winsound.Beep(500, 500)
cls()
print(f"hi, {name}!")
winsound.Beep(1100, 450)
winsound.Beep(1300, 700)
while True:
    version = "0.020"
    cmd = input(Fore.YELLOW +"jungua" + Fore.GREEN + "@" + Fore.WHITE + f"{name}: ")
    winsound.Beep(1500, 200)
    if cmd == 'whoami':
        print(Fore.YELLOW + "your username: " + Fore.WHITE + f"{name}")
    elif cmd.lower() == 'exit':
        break
    elif cmd.lower() == 'fetch':
        disk = psutil.disk_usage('C:\\' if psutil.WINDOWS else '/')
        diskgb = round(disk.total / (1024**3), 2)
        ram = psutil.virtual_memory()
        ramgb = round(ram.total / (1024**3), 2)
        clocker = time.strftime("%Y-%m-%d %H:%M:%S")
        print(Fore.YELLOW + """
             ###########
              #########
              #########
             ###########
              
             ###########
              #########
              #########
              #########
              #########
    ######    #########
   ######## ##########
     ################
        #########
      
""" + Fore.WHITE)
        print(Fore.YELLOW + "version: " + Fore.WHITE + f"{version}\n" + Fore.YELLOW + "disk storage: " + Fore.WHITE + f"{diskgb}GB\n" + Fore.YELLOW + "ram storage: " + Fore.WHITE + f"{ramgb}GB\n" + Fore.YELLOW + "time now: " + Fore.WHITE + f"{clocker}")
    elif cmd.lower() == 'clear' or cmd.lower() =='cls':
        cls()
    elif cmd.startswith('say(') and cmd.endswith(')'):
        text = cmd[4:-1]
        print(Fore.GREEN + f"{text}" + Fore.WHITE)
    elif cmd.lower() == 'rename':
        threading.Thread(target=beeep, daemon=True).start()
        newname = input(Fore.GREEN +"rename: " + Fore.WHITE)
        if not newname:
            print(Fore.RED + "err: " + Fore.WHITE + "username not defined")
            winsound.Beep(500, 500)
            continue
        threading.Thread(target=beep, daemon=True).start()
        aquest = input(Fore.RED + "do you really want to rename?\n" + Fore.YELLOW +"before rename: " + Fore.WHITE + f"{name}\n" + Fore.YELLOW + "after rename: " + Fore.WHITE + f"{newname}\n" + Fore.LIGHTBLACK_EX + "(y/n) " + Fore.WHITE)
        if aquest.lower() == 'y':
            if newname == name:
                print(Fore.YELLOW + "info: " + Fore.WHITE + "this is already your name")
                winsound.Beep(800, 300)
                continue
            else:
                write(newname)
                name = newname
                winsound.Beep(1000, 300)  
                winsound.Beep(1200, 300)
        else:
            print(Fore.YELLOW + "rename cancelled" + Fore.WHITE)
            winsound.Beep(600, 300)
            pass
    elif cmd.lower() == "help":
        threading.Thread(target=beep, daemon=True).start()
        print(Fore.RED + "check" + Fore.YELLOW + " 'README.md' " + Fore.RED + "on official repository: " + Fore.LIGHTBLACK_EX + "https://github.com/arxxdd-scheg/jungua" + Fore.WHITE)